backend-api/template.yaml

```diff
@@ -243,6 +243,95 @@ Resources:
           Parameters:
             Location: ./openApiSpecs/async-private-spec.yaml
 
+  PrivateApiDomainName:
+    Type: AWS::ApiGateway::DomainName
+    Properties:
+      DomainName: !Sub
+        - private-${AWS::StackName}.${DNS_RECORD}
+        - DNS_RECORD: !FindInMap
+            - DNS
+            - !Ref Environment
+            - BaseDns
+      EndpointConfiguration:
+        Types:
+          - REGIONAL
+      RegionalCertificateArn: !Sub '{{resolve:ssm:/${Environment}/Platform/ACM/AsyncPrimaryZoneWildcardCertificateARN}}' # Uses a wildcard (in dev only?)
+      SecurityPolicy: TLS_1_2
+
+  PrivateApiBasePathMapping:
+    Type: AWS::ApiGateway::BasePathMapping
+    DependsOn: PrivateApiDomainName
+    Properties:
+      DomainName: !Sub
+        - private-${AWS::StackName}.${DNS_RECORD}
+        - DNS_RECORD: !FindInMap
+            - DNS
+            - !Ref Environment
+            - BaseDns
+      RestApiId: !Ref PrivateApi
+      Stage: !Ref PrivateApi.Stage
+
+  PrivateApiRecordSet:
+    Type: AWS::Route53::RecordSet
+    Properties:
+      Name: !Sub
+        - private-${AWS::StackName}.${DNS_RECORD}
+        - DNS_RECORD: !FindInMap
+            - DNS
+            - !Ref Environment
+            - BaseDns
+      Type: CNAME
+      HostedZoneId: !Sub '{{resolve:ssm:/${Environment}/Platform/Route53/PrimaryZoneID}}'
+      TTL: 60
+      ResourceRecords:
+        - !GetAtt PrivateApiLoadBalancer.DNSName # this will be the load balancer domain
+
+  PrivateApiLoadBalancer:
+    Type: AWS::ElasticLoadBalancingV2::LoadBalancer
+    Properties:
+      IpAddressType: ipv4
+      Scheme: internal
+      Subnets:
+        - !ImportValue devplatform-vpc-PrivateSubnetIdA
+        - !ImportValue devplatform-vpc-PrivateSubnetIdB
+        - !ImportValue devplatform-vpc-PrivateSubnetIdC
+      Type: network
+
+  PrivateApiListener:
+    Type: AWS::ElasticLoadBalancingV2::Listener
+    Properties:
+      DefaultActions:
+        - Order: 1
+          TargetGroupArn: !Ref PrivateApiTargetGroup
+          Type: forward
+      LoadBalancerArn: !Ref PrivateApiLoadBalancer
+      Port: 443
+      Certificates:
+        - CertificateArn: !Sub '{{resolve:ssm:/${Environment}/Platform/ACM/AsyncPrimaryZoneWildcardCertificateARN}}'
+      Protocol: TLS
+
+  PrivateApiTargetGroup:
+    Type: AWS::ElasticLoadBalancingV2::TargetGroup
+    Properties:
+      HealthCheckEnabled: true
+      HealthCheckIntervalSeconds: 10
+      HealthCheckPort: 443
+      HealthCheckProtocol: TCP
+      HealthCheckTimeoutSeconds: 10
+      HealthyThresholdCount: 3
+      UnhealthyThresholdCount: 3
```

Comment:
are the targets able to service the health check?

Reply:
The targets are reporting healthy at the moment 👍  

![image](https://github.com/user-attachments/assets/08dd35c7-f953-43d5-84f1-d1375a4f6dc8)